node/src/Pos/Wallet/Web/Pending/Functions.hs

```diff
@@ -0,0 +1,174 @@
+import           Pos.Crypto                   (WithHash (..), hash)
+import           Pos.Core.Configuration       (HasConfiguration, genesisHash)
+import           Pos.Core.Types               (ChainDifficulty)
+import           Pos.Core.Class               (difficultyL, headerHash, prevBlockL)
+import qualified Pos.DB.DB                    as DB
+import qualified Pos.DB.Block                 as DB
+import           Pos.Txp.Core                 (Tx, TxId, topsortTxs, taTx, txpTxs)
+import           Pos.Txp.Toil.Class           (MonadUtxoRead)
+import           Pos.Txp.Toil.Trans           (evalToilTEmpty)
+import           Pos.Txp.Toil.Failure         (ToilVerFailure)
+import           Pos.Txp.Toil.Utxo.Functions  (VTxContext (..), verifyTxUtxo,
+                                               applyTxToUtxo)
+import           Pos.Wallet.Web.Pending.Types (PendingTx (..), PtxCondition (..),
+                                               ptxCond, _PtxApplying, _PtxWontApply,
+                                               ptxTxAux, ptxTxId)
+import           Pos.Util.Util                (getKeys)
+
+
+-- | Reevaluate all pending txs (in 'PtxApplying' and 'PtxWontApply' states):
+--
+--   * 'PtxInNewestBlocks' for ones that are already in the blockchain
+--     (at any depth)
+--   * 'PtxWontApply' for ones that can't be applied
+--   * 'PtxApplying' to ones that can be applied either directly or after
+--     applying some other transactions
+--
+-- TODO: do we need to do anything special to ensure that the UTXO we're
+-- shown doesn't include outputs from failed txs or mempool txs? I'm not
+-- sure
+reevaluateUncertainPtxs
+    :: forall ssc m.
+       (DB.MonadBlockDB ssc m, HasConfiguration, MonadUtxoRead m, WithLogger m)
+    => HashMap TxId PendingTx -> m (HashMap TxId PendingTx)
+reevaluateUncertainPtxs ptxs = do
+    -- Get all transactions we're uncertain about
+    let isUncertain ptx = has (ptxCond . _PtxApplying)  ptx ||
+                          has (ptxCond . _PtxWontApply) ptx
+    let (uncertain, certain) = partitionHashMap isUncertain ptxs
+    -- Find those that are already in blocks; they'll be marked as
+    -- 'PtxInNewestBlocks'
+    presentIds <- getTxsBlockStatus @ssc (getKeys uncertain)
+    let present :: HashMap TxId (PendingTx, ChainDifficulty)
+        present = HM.intersectionWith (,) uncertain presentIds
+    let missing :: HashMap TxId PendingTx
+        missing = HM.difference uncertain presentIds
+    -- Topsort the rest ('missing'), go one by one and either add to UTXO or
+    -- mark as 'PtxWontApply'
+    --
+    -- TODO: is evalToilTEmpty okay?
+    (missingInvalid, missingValid) <- evalToilTEmpty $ do
+        let sorted = fromMaybe (toList missing) $
+                     topsortTxs ptxWithHash (toList missing)
+        fmap partitionEithers $ forM sorted $ \ptx -> do
+            let vtc = VTxContext True     -- TODO: is it necessarily 'True'?
```

Comment:
In scope of this hotfix yes. If we want to have this endpoint in `develop`, we should revise this line before porting it to `develop`.